lib/src/deploy.rs

```diff
@@ -340,6 +347,38 @@ pub(crate) fn switch_origin_inplace(root: &Dir, imgref: &ImageReference) -> Resu
     Ok(newest_deployment)
 }
 
+pub fn get_kargs(repo: &ostree::Repo, fetched: &ImageState) -> Result<Vec<String>> {
+    let cancellable = gio::Cancellable::NONE;
+    let (fetched_tree, _) = repo.read_commit(fetched.ostree_commit.as_str(), cancellable)?;
+    let fetched_tree = fetched_tree.resolve_relative_path("/usr/lib/bootc/kargs.d");
```

Comment:
Yeah, the recursion here is unfortunate...not a blocker but perhaps it is an argument for driving this logic into ostree.

Or alternatively I guess we'd need to split up the ostree deployment phase into:

- check out filesystem tree
- compute data from it
- set up bootloader entry

Which I believe exists in the API.